src/raster_programs.rs

```diff
@@ -0,0 +1,72 @@
+use crate::{errors::*, Dataset};
+use libc::{c_char, c_int};
+use std::ffi::CString;
+use std::{
+    borrow::Borrow,
+    ptr::{null, null_mut},
+};
+
+/// Build a VRT from a list of datasets.
+/// Wraps [GDALBuildVRT].
+/// See the [program docs] for more details.
+///
+/// [GDALBuildVRT]: https://gdal.org/api/gdal_utils.html#gdal__utils_8h_1a057aaea8b0ed0476809a781ffa377ea4
+/// [program docs]: https://gdal.org/programs/gdalbuildvrt.html
+pub fn build_vrt<D: Borrow<Dataset>>(
+    dest: Option<String>,
+    datasets: &[D],
+    args: Vec<String>,
+) -> Result<Dataset> {
```

Comment:
It might be nicer to split this function into generic and non-generic parts, like in https://doc.rust-lang.org/src/std/path.rs.html#1237-1239.